pkg/nsx/services/inventory/build.go

```diff
@@ -56,6 +56,26 @@ func (s *InventoryService) BuildPod(pod *corev1.Pod) (retry bool) {
 		status = InventoryStatusUnknown
 	}
 
+	// Create network errors from pod conditions and status message
+	var networkErrors []common.NetworkError
+	if pod.Status.Phase != corev1.PodRunning {
+		// Check for a message in pod status
+		if pod.Status.Message != "" {
+			networkErrors = append(networkErrors, common.NetworkError{
+				ErrorMessage: pod.Status.Message,
+			})
+		}
+
+		// Check conditions
+		for _, condition := range pod.Status.Conditions {
+			if condition.Status == corev1.ConditionFalse {
```

Comment:
Why we collect all the condition message and sorted them for node but just false condition message for Pod?
https://github.com/vmware-tanzu/nsx-operator/blob/6095f84246ed681b215ea32852a806432bdd2cb0/pkg/nsx/services/inventory/build.go#L427-L435

Reply:
![image](https://github.com/user-attachments/assets/8f332494-316f-4cbe-a9ee-7e3df8c1881a)
Pod is different from VM, the status message carries more significance in this context.